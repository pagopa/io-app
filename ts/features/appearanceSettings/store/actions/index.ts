/**
 * Action types and action creator related to the Appearance settings.
 */
import { ActionType, createStandardAction } from "typesafe-actions";

export const setShowAppearanceSettingsBanner = createStandardAction(
  "SHOW_APPEARANCE_SETTINGS_BANNER"
)<boolean>();

export type AppearanceSettingsActions = ActionType<
  typeof setShowAppearanceSettingsBanner
>;
