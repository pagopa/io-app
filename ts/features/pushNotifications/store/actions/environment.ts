import { ActionType, createStandardAction } from "typesafe-actions";

export const updateSystemNotificationsEnabled = createStandardAction(
  "UPDATE_SYSTEM_NOTIFICATIONS_ENABLED"
)<boolean>();
export const setPushPermissionsRequestDuration = createStandardAction(
  "SET_PUSH_PERMISSIONS_REQUEST_DURATION"
)<number>();

export type NotificationPermissionsActions =
  | ActionType<typeof updateSystemNotificationsEnabled>
  | ActionType<typeof setPushPermissionsRequestDuration>;
