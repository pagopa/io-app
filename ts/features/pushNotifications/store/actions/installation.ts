import { ActionType, createStandardAction } from "typesafe-actions";

export const newPushNotificationsToken = createStandardAction(
  "NOTIFICATIONS_INSTALLATION_TOKEN_UPDATE"
)<string>();

// the notification token is registered in the backend
export const pushNotificationsTokenUploaded = createStandardAction(
  "NOTIFICATIONS_INSTALLATION_TOKEN_REGISTERED"
)<string>();

export type NotificationsActions =
  | ActionType<typeof newPushNotificationsToken>
  | ActionType<typeof pushNotificationsTokenUploaded>;
