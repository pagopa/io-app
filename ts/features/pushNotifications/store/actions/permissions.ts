import { ActionType, createStandardAction } from "typesafe-actions";

export const updateSystemNotificationsEnabled = createStandardAction(
  "UPDATE_SYSTEM_NOTIFICATIONS_ENABLED"
)<boolean>();

export type NotificationPermissionsActions = ActionType<
  typeof updateSystemNotificationsEnabled
>;
