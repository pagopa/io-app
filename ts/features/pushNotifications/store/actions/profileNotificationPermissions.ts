import { ActionType, createStandardAction } from "typesafe-actions";

export const notificationsInfoScreenConsent = createStandardAction(
  "NOTIFICATIONS_INFO_SCREEN_CONSENT"
)<void>();

export type ProfileNotificationPermissionsActions = ActionType<
  typeof notificationsInfoScreenConsent
>;
