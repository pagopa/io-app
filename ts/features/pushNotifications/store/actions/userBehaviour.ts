import { ActionType, createStandardAction } from "typesafe-actions";

export const setEngagementScreenShown = createStandardAction(
  "SET_ENGAGEMENT_SCREEN_SHOWN"
)<void>();

export type UserBehaviourActions = ActionType<typeof setEngagementScreenShown>;
