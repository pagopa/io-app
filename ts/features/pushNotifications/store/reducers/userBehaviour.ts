import { getType } from "typesafe-actions";
import { Action } from "../../../../store/actions/types";
import { setEngagementScreenShown } from "../actions/userBehaviour";

export type UserBehaviourState = {
  engagementScreenShown: boolean;
};

export const INITIAL_STATE: UserBehaviourState = {
  engagementScreenShown: false
};

export const userBehaviourReducer = (
  state: UserBehaviourState = INITIAL_STATE,
  action: Action
): UserBehaviourState => {
  switch (action.type) {
    case getType(setEngagementScreenShown):
      return { ...state, engagementScreenShown: true };
  }
  return state;
};
