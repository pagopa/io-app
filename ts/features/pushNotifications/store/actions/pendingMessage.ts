import { ActionType, createStandardAction } from "typesafe-actions";
import { PendingMessageState } from "../reducers/pendingMessage";

export const updateNotificationsPendingMessage = createStandardAction(
  "NOTIFICATIONS_PENDING_MESSAGE_UPDATE"
)<PendingMessageState>();

export const clearNotificationPendingMessage = createStandardAction(
  "NOTIFICATIONS_PENDING_MESSAGE_CLEAR"
)<void>();

export type PendingMessageActions =
  | ActionType<typeof updateNotificationsPendingMessage>
  | ActionType<typeof clearNotificationPendingMessage>;
