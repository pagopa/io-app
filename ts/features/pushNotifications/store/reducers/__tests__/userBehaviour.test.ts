import { applicationChangeState } from "../../../../../store/actions/application";
import { setEngagementScreenShown } from "../../actions/userBehaviour";
import { INITIAL_STATE, userBehaviourReducer } from "../userBehaviour";

describe("userBehaviourReducer", () => {
  it("INITIAL_STATE should match expected values", () => {
    expect(INITIAL_STATE.engagementScreenShown).toBe(false);
  });
  it("Should match snapshot", () => {
    const userBehaviourState = userBehaviourReducer(
      undefined,
      applicationChangeState("active")
    );
    expect(userBehaviourState).toMatchSnapshot();
  });
  it("'engagementScreenShown' should be 'true' after receiving 'setEngagementScreenShown'", () => {
    const userBehaviourState = userBehaviourReducer(
      undefined,
      setEngagementScreenShown()
    );
    expect(userBehaviourState.engagementScreenShown).toBe(true);
  });
});
