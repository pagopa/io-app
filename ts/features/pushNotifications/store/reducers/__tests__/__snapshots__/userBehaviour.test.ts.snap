// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userBehaviourReducer Should match snapshot 1`] = `
{
  "engagementScreenShown": false,
}
`;
